apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -163,12 +184,20 @@
         %% shard (if applicable to the backend), as the batch will be split accordingly
         %% even if this flag is `true'.
         %% Default: `false'.
-        atomic => boolean()
+        atomic => boolean() | isolate
```

Comment:
What is semantic of `atomic=>isolate` and how is it different from `atomic => true`?

Reply:
In essence,
* `atomic => true` means that either whole set of operations is applied or none at all,
* `atomic => isolate` means the same + that set of operations applied in isolation from any other set, basically to be sure that any failing precondition will affect only this set of operations.

Admittedly, backends can decide on their own that some level of isolation is needed if there are preconditions in a batch, so this explicitness is not necessary.

Reply:
I think `atomic => true` should do what `atomic => isolate` does.